interface/operations/operations_client.go

```diff
@@ -155,7 +161,7 @@ func (c Client) ListKeys() ([]*listkeys.KeyInfo, error) {
 func (c Client) ListAuthenticators() ([]*listauthenticators.AuthenticatorInfo, error) {
 	req := &listauthenticators.Operation{}
 	resp := &listauthenticators.Result{}
-	err := c.operation(requests.OpListKeys, requests.ProviderCore, req, resp)
+	err := c.operation(requests.OpListAuthenticators, requests.ProviderCore, req, resp)
```

Comment:
assume you mean not take a provider?  Need the opcode to be different per operation.

Reply:
ah yes sorry, "not take a provider ID" is what I meant

Reply:
Good idea, but when got into detail, was going to be quite messy to do it with the existing provider field of the operations.Client.

ended up using a strategy pattern using the operations.ProviderSelector interface to allow the operations.Client to call back to the caller (parsec.BasicClient) to get the best provider to use.  Also moved the provider crypto checks to parsec.BasicClient

Reply:
Arg, what kind of details? Sorry if I was confusing. I was thinking of something dead simple:
Keeping the `provider` and `authType` fields on `Client` and modifying `InitClient` to:
```go
func InitClient(provider requests.ProviderID, authType auth.AuthenticationType) (*Client, error) {
	conn, err := connection.NewDefaultConnection()
	if err != nil {
		return nil, err
	}
	client := &Client{
		conn:     conn,
		provider: provider,
		authType: authType,
	}

	return client, nil
}
```
and in the `operation` function:
```go
r, err := requests.NewRequest(op, request, authenticator, c.provider)
```

So that we leave all the logic of automatic provider and authenticator selection to the Basic Client, in a simple way? Sorry if that does not work still!!